src/components/AutoComplete/AutoComplete.js

```diff
@@ -0,0 +1,127 @@
+import AutoCompleteInput from './AutoCompleteInput';
+import AutoCompleteItems from './AutoCompleteItems';
+import { KEY_CODES } from '../../common/helpers';
+
+export const getActiveItems = items =>
+  items
+    .filter(
+      ({ disabled, type }) => !disabled && !['header', 'divider'].includes(type)
+    )
+    .map(({ text }) => text);
+
+class AutoComplete extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      inputValue: ''
+    };
+  }
+
+  render() {
+    const {
+      onSearch,
+      labelText,
+      onInputUpdate,
+      items,
+      actionText,
+      ...rest
+    } = this.props;
+
+    const activeItems = getActiveItems(items);
+
+    return (
+      <Downshift
+        onStateChange={({ inputValue, ...rest }) => {
+          if (typeof inputValue === 'string') {
+            onInputUpdate(inputValue);
+            this.setState({ inputValue });
+          }
+        }}
+        defaultHighlightedIndex={0}
+        selectedItem={this.state.inputValue}
+        {...rest}
+      >
+        {({
+          getInputProps,
+          getItemProps,
+          getLabelProps,
+          isOpen,
+          inputValue,
+          highlightedIndex,
+          getRootProps,
+          selectedItem,
+          selectItem
+        }) => (
+          <div>
+            {labelText && <label {...getLabelProps()}>{labelText}</label>}
+            <InputGroup>
+              <AutoCompleteInput
+                onKeyPress={e => {
+                  switch (e.keyCode) {
+                    case KEY_CODES.TAB_KEY:
+                      if (isOpen && activeItems[highlightedIndex]) {
+                        selectItem(activeItems[highlightedIndex]);
+                        e.preventDefault();
+                      }
+
+                      break;
+
+                    case KEY_CODES.ENTER_KEY:
+                      if (!isOpen || !activeItems[highlightedIndex]) {
+                        onSearch(this.state.inputValue);
+                        e.preventDefault();
+                      }
+
+                      break;
+
+                    default:
+                      break;
+                  }
+                }}
+                passedProps={getInputProps()}
+              />
+              <InputGroup.Button>
+                <Button onClick={() => onSearch(inputValue)}>
+                  {actionText}
+                </Button>
+              </InputGroup.Button>
+            </InputGroup>
+
+            {isOpen && items.length ? (
+              <AutoCompleteItems
+                {...{
+                  items,
+                  highlightedIndex,
+                  selectedItem,
+                  getItemProps,
+                  activeItems
+                }}
+              />
+            ) : null}
+          </div>
+        )}
+      </Downshift>
+    );
+  }
+}
+
+AutoComplete.propTypes = {
+  items: PropTypes.arrayOf(PropTypes.object).isRequired,
```

Comment:
Can you also add a description comment?
Checkout the `Alert` propTypes:
https://github.com/patternfly/patternfly-react/blob/master/src/components/Alert/Alert.js#L38

And how they compiled into the storybook:
https://rawgit.com/patternfly/patternfly-react/gh-pages/index.html?selectedKind=Alert&selectedStory=Alert%20types&full=0&addons=1&stories=1&panelRight=0&addonPanel=storybooks%2Fstorybook-addon-knobs
Click "show info" and scroll down to the prop types description.

Reply:
Wow that's cool! Thanks for sharing 👍  Added them in now.